frontend/src/components/ui/Button/button.jsx

```diff
@@ -0,0 +1,207 @@
+        active:tw-bg-button-danger-secondary-pressed tw-disabled:tw-border-border-default
+        tw-disabled:tw-bg-button-danger-secondary-disabled
+        tw-disabled:tw-text-text-disabled tw-focus-visible:tw-interactive-focus-outline tw-interactive-focus-nonsolid`,
+    },
+    size: {
+      large: `tw-h-[40px] tw-gap-[8px] tw-py-[10px] tw-rounded-[10px] tw-text-lg`,
+      default: `tw-h-[32px] tw-gap-[6px] tw-py-7px] tw-rounded-[8px] tw-text-base`,
+      medium: `tw-h-[28px] tw-gap-[6px] tw-py-[5px] tw-rounded-[6px] tw-text-base`,
+      small: `tw-h-[20px] tw-gap-[4px] tw-py-[2px] tw-rounded-[4px] tw-text-sm`,
+    },
+  },
+  //this part will be updated/improved
+  compoundVariants: [
+    {
+      iconOnly: true,
+      size: 'large',
+      className: 'tw-w-[40px] tw-px-[10px]',
+    },
+    {
+      iconOnly: true,
+      size: 'default',
+      className: 'tw-w-[32px] tw-px-[7px]',
+    },
+    {
+      iconOnly: true,
+      size: 'medium',
+      className: 'tw-w-[28px] tw-px-[5px]',
+    },
+    {
+      iconOnly: true,
+      size: 'small',
+      className: 'tw-w-[20px] tw-px-[2px]',
+    },
+    {
+      iconOnly: false,
+      size: 'large',
+      className: 'tw-px-[20px]',
+    },
+    {
+      iconOnly: false,
+      size: 'default',
+      className: 'tw-px-[12px]',
+    },
+    {
+      iconOnly: false,
+      size: 'medium',
+      className: 'tw-px-[10px]',
+    },
+    {
+      iconOnly: false,
+      size: 'small',
+      className: 'tw-px-[8px]',
+    },
+  ],
+  defaultVariants: {
+    variant: 'primary',
+    size: 'default',
+  },
+});
+
+const Button = forwardRef(
+  (
+    {
+      className,
+      variant = 'primary',
+      size = 'default',
+      leadingIcon,
+      trailingIcon,
+      isLoading,
+      disabled,
+      asChild = false,
+      fill = '',
+      iconOnly = false, // as normal button and icon have diff styles make sure to pass it as truw when icon only button is used
+      ...props
+    },
+    ref
+  ) => {
+    const iconFillColor = !defaultButtonFillColour.includes(fill) && fill ? fill : getDefaultIconFillColor(variant);
+    const Comp = asChild ? Slot : 'button';
+    const leadingIconElement = leadingIcon && (
+      <SolidIcon name={leadingIcon} height={getIconSize(size)} width={getIconSize(size)} fill={iconFillColor} />
+    );
+    const trailingIconElement = trailingIcon && (
+      <SolidIcon name={trailingIcon} height={getIconSize(size)} width={getIconSize(size)} fill={iconFillColor} />
+    );
+
+    return (
+      <Comp
+        className={cn(buttonVariants({ variant, size, iconOnly, className }))}
+        ref={ref}
+        disabled={disabled}
+        {...props}
+      >
+        {isLoading ? (
+          <div className="tw-flex tw-justify-center tw-items-center">
+            <Loader color={iconFillColor} width={getIconSize(size)} />
+            <a className="tw-invisible">{props.children}</a>
```

Comment:
Are we adding this to maintain the width?

Reply:
Yes , for getting icon height and width